rclcpp_action/include/rclcpp_action/client.hpp

```diff
@@ -283,8 +283,11 @@ class Client : public ClientBase
 
     /// Function called when the goal is accepted or rejected.
     /**
-     * If the goal is accepted, then a shared pointer to a valid goal handle is passed as the
-     * single argument. If the goal is rejected, then pointer has the value `nullptr`.
+     * Takes a single argument that is a future to a goal handle shared pointer.
+     * If the goal is accepted, then the pointer points to a valid goal handle.
+     * If the goal is rejected, then pointer has the value `nullptr`.
+     * If an error occurs while waiting for the goal response an exception will be thrown
+     * when calling `future::get()`.
```

Comment:
It would be nice to list the at least some of the exceptions we know it can throw here.

Reply:
Done b69a1fd